sotodlib/site_pipeline/update_det_cal.py

```diff
@@ -0,0 +1,838 @@
+                if r.chdata.band == band and r.chdata.channel == chan:
+                    return r
+            return None
+
+        for i, cal in enumerate(cals):
+            band = am.det_info.smurf.band[i]
+            chan = am.det_info.smurf.channel[i]
+            detset = am.det_info.detset[i]
+            stream_id = am.det_info.stream_id[i]
+            bg = cal.bg
+            bsa = bsas[detset]
+
+            if bsa is None or bg == -1:
+                continue
+
+            bl_label = f"{stream_id}_b{bg:0>2}"
+            if not bias_line_is_valid[bl_label]:
+                continue
+
+            ridx = np.where((bsa.bands == band) & (bsa.channels == chan))[0]
+            if not ridx:  # Channel doesn't exist in bias step analysis
+                continue
+
+            if cfg.apply_cal_correction:
+                correction = find_correction_results(band, chan, detset)
+                if correction is None:
+                    logger.warn(
+                        "Unable to find correction result for %s %s %s (%s)",
+                        band,
+                        chan,
+                        detset,
+                        obs_id,
+                    )
+                    use_correction = False
+                    cal.tes_param_correction_success = False
+                else:
+                    use_correction = correction.success
+                    cal.tes_param_correction_success = correction.success
+            else:
+                use_correction = False
+
+            ridx = ridx[0]
+            cal.tau_eff = bsa.tau_eff[ridx]
+            if bg != -1:
+                cal.v_bias = bsa.Vbias[bg]
+
+            if use_correction and correction.corrected_params is not None:
+                cpars = correction.corrected_params
+                cal.r_tes = cpars.corrected_R0
+                cal.r_frac = cpars.corrected_R0 / cal.r_n
+                cal.s_i = cpars.corrected_Si * 1e6
+                cal.p_bias = cpars.corrected_Pj * 1e-12
+                cal.loopgain = cpars.loopgain
+            else:
+                cal.r_tes = bsa.R0[ridx]
+                cal.r_frac = bsa.Rfrac[ridx]
+                cal.p_bias = bsa.Pj[ridx]
+                cal.s_i = bsa.Si[ridx]
+
+            # Save naive parameters even if we're using corrected version
+            cal.naive_r_tes = bsa.R0[ridx]
+            cal.naive_r_frac = bsa.Rfrac[ridx]
+            cal.naive_s_i = bsa.Si[ridx]
+            cal.naive_p_bias = bsa.Pj[ridx]
+
+            if cal.s_i == 0:
+                cal.phase_to_pW = np.nan
+            else:
+                cal.phase_to_pW = pA_per_phi0 / (2 * np.pi) / cal.s_i * cal.polarity
+
+        res.result_set = np.array([astuple(c) for c in cals], dtype=CalInfo.dtype())
+        res.success = True
+    except Exception as e:
+        res.traceback = traceback.format_exc()
+        res.fail_msg = res.traceback
+        if cfg.raise_exceptions:
+            raise
```

Comment:
`raise e` would be better here, right, because the user would get some kind of informative traceback?

Reply:
`raise` by itself re-raises the current exception without modifying the stack-trace. [Looking into it there's a slight difference between the two in how the trace is set](https://stackoverflow.com/questions/36153805/difference-between-raise-and-raise-e), but it looks like it doesn't really matter for python3.

Reply:
Ah, I hadn't appreciated the subtleties.  I think `raise e` is still slightly better.  It's helpful to know both where the error came from and that it was caught and re-raised.